src/circuits/query.ts

```diff
@@ -21,59 +23,65 @@ const serializationValueDataSlotBType = 'serialization:ValueDataSlotB';
 
 // Query is a query to circuit
 export interface Query {
-  allowedIssuers: string[];
+  allowedIssuers: string;
   req: Map<string, unknown>;
-  schema: Schema;
-  claimId: string;
+  context: string;
+  type: string;
+  claimID?: string;
 }
 
 // ClaimOutputs fields that are used in proof generation
 export interface ClaimOutputs {
-  issuerId: string;
-  schemaHash: bigint;
-  slotIndex: number;
+  issuerId: Id;
+  schemaHash: SchemaHash;
+  slotIndex?: number;
   operator: number;
   value: bigint[];
+  timestamp: number;
+  merklized: number;
+  claimPathKey?: bigint;
+  claimPathNotExists?: number;
+  valueArraySize: number;
 }
 
 export async function checkQueryRequest(
   query: Query,
   outputs: ClaimOutputs,
   schemaLoader: ISchemaLoader,
 ): Promise<void> {
-  const issuerAllowed = query.allowedIssuers.some(
-    (issuer) => issuer === '*' || issuer === outputs.issuerId,
-  );
-  if (!issuerAllowed) {
+  if (
+    query.allowedIssuers !== '*' &&
+    query.allowedIssuers !== outputs.issuerId.string()
+  ) {
     throw new Error('issuer is not in allowed list');
   }
 
   let loadResult: SchemaLoadResult;
 
   try {
-    loadResult = await schemaLoader.load(query.schema);
+    loadResult = await schemaLoader.load(query.context);
   } catch (e) {
     throw new Error(`can't load schema for request query`);
   }
 
-  if (loadResult.extension !== 'json-ld') {
-    throw new Error('only json-ld schema is supported');
+  const schemaHash = CreateSchemaHash(query.context, query.type);
+  if (schemaHash.toString() !== outputs.schemaHash.toString()) {
+    throw new Error(`schema that was used is not equal to requested in query`);
   }
-  const toHash = new Uint8Array([
-    ...loadResult.schema,
-    ...toBytes(query.schema.type),
-  ]);
 
-  const schemaHash = keccak256(Buffer.from(toHash));
-
-  // only json ld-schema are supported
-  const cq = parseRequest(
+  const cq = await parseRequest(
     query.req,
     loadResult.schema,
-    query.schema.type,
+    query.context,
+    query.type,
     outputs.value.length,
+    outputs.merklized,
   );
 
+  if (query.req === undefined) {
```

Comment:
could it be replaced to if (!query.req) ?

Reply:
Done